pkg/policy/l4.go

```diff
@@ -132,6 +132,31 @@ type L4Filter struct {
 	policy unsafe.Pointer // *L4Policy
 }
 
+// CopyL7RulesPerEndpoint creates a copy of the L7RulesPerEp of the L4Filter.
+func (l4 *L4Filter) CopyL7RulesPerEndpoint() L7DataMap {
```

Comment:
just a warning this is a not a deepcopy.

Reply:
👍  - the old behavior did exactly this, just inline in `redirect.updateRules`:

```
@@ -73,12 +73,9 @@ func newRedirect(localEndpoint logger.EndpointUpdater, listener *ProxyPort, dstP
 // updateRules updates the rules of the redirect, Redirect.mutex must be held
 // 'implementation' is not initialized when this is called the first time.
 // TODO: Replace this with RedirectImplementation UpdateRules method!
-func (r *Redirect) updateRules(l4 *policy.L4Filter) revert.RevertFunc {
+func (r *Redirect) updateRules(p policy.ProxyPolicy) revert.RevertFunc {
        oldRules := r.rules
-       r.rules = make(policy.L7DataMap, len(l4.L7RulesPerEp))
-       for key, val := range l4.L7RulesPerEp {
-               r.rules[key] = val
-       }
+       r.rules = p.CopyL7RulesPerEndpoint()
        return func() error {
                r.mutex.Lock()
                r.rules = oldRules
```

so there is no functional change here.